borsh/src/ser/mod.rs

```diff
@@ -267,6 +268,12 @@ where
 {
     #[inline]
     fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        if size_of::<T>() == 0 {
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                "Zero-sized types are not allowed.",
```

Comment:
I feel this might be more helpful to the developers:

```suggestion
                "Vectors of zero-sized types are not allowed due to deny-of-service concerns on deserialization.",
```